apps/emqx_durable_storage/src/emqx_ds_optimistic_tx.erl

```diff
@@ -556,18 +573,27 @@ do_cook_dirty_appends(DB, Shard, CBM, Gen, GenData, PresumedCommitSerial, Pendin
 This function must be called after adding a new item pending for commit.
 It (re)schedules flushing of the data.
 """.
--spec finalize_add_pending(leader_state(), d(), [gen_statem:action()]) ->
-    leader_loop_result().
-finalize_add_pending(_, D = #d{n_items = N, max_items = MaxItems}, Actions) when
-    N >= MaxItems, MaxItems > 0
-->
-    handle_flush(D, Actions);
-finalize_add_pending(?idle, D = #d{idle_flush_interval = IdleInterval}, Actions) ->
-    %% Schedule early flush if the shard is idle:
-    Timeout = {timeout, IdleInterval, ?timeout_flush},
-    {next_state, ?leader(?pending), D, [Timeout | Actions]};
-finalize_add_pending(_, D, Actions) ->
-    {keep_state, D, Actions}.
+-spec finalize_add_pending(d()) -> d().
+finalize_add_pending(
+    D0 = #d{
+        entered_pending_at = EP, flush_interval = FlushInterval, n_items = N, max_items = MaxItems
+    }
+) ->
+    D =
+        case EP of
+            _ when is_integer(EP) ->
+                D0;
+            undefined ->
+                %% Schedule unconditional flush after FlushInterval:
+                erlang:send_after(FlushInterval, self(), ?timeout_flush),
```

Comment:
Nit. Perhaps save on setting up an ERTS timer if `FlushInterval` is 0, and just send a message? As 0 is the default now.